### PR TITLE
windows: greatly simplify developer system setup.

### PR DESCRIPTION
**Pull request by mlyle**
As documented (prematurely) on
https://github.com/TauLabs/TauLabs/wiki/Development-Windows-setup

- Create new bash_profile that finds tools appropriately
- Remove legacy winx86 stuff that isn't used.
- Update tools.mk to use curl to install openssl and use that in the
  directions.

### CODE REVIEW

**Comment by mlyle** [comment]
For #1789.

File: make/tools.mk [hunk 1]
```diff
@@ -565,15 +565,14 @@ endif
...
+  openssl_install: OPENSSL_URL  := https://slproweb.com/download/Win32OpenSSL-1_0_2d.exe
 
 openssl_install: OPENSSL_FILE := $(notdir $(OPENSSL_URL))
 OPENSSL_DIR = $(TOOLS_DIR)/win32openssl
 # order-only prereq on directory existance:
 openssl_install : | $(DL_DIR) $(TOOLS_DIR)
 openssl_install: openssl_clean
-        # download the instalatopn file only if it's newer than what we already have
-	$(V1) wget -N -P "$(DL_DIR)" "$(OPENSSL_URL)"
+	$(V1) curl -L -k -o "$(DL_DIR)/$(OPENSSL_FILE)" "$(OPENSSL_URL)"
```
Comment (kubark42) [inline make/tools.mk]: Is ```curl``` commonly available on Windows? Not that I mind, getting ```wget``` was so much harder than it should have been. It was like the ```wget``` maintainers actively didn't want anyone easily finding and installing it.

**Comment by kubark42** [comment]
Oh.

My.

God.

Wonderful, wunderbar, fantastique, thanks so much!!!

File: make/tools.mk [hunk 2]
```diff
@@ -565,15 +565,14 @@ endif
 
 # OPENSSL download URL
 ifdef WINDOWS
-  openssl_install: OPENSSL_URL  := http://slproweb.com/download/Win32OpenSSL-1_0_2c.exe
+  openssl_install: OPENSSL_URL  := https://slproweb.com/download/Win32OpenSSL-1_0_2d.exe
```
Comment (kubark42) [inline make/tools.mk]: This should go in another PR. Or at least in another commit. I want to push hard on making commits and PRs as atomic as possible.

File: make/winx86/cmd/sh.cmd [hunk 3]
```diff
@@ -1,135 +0,0 @@
...
-
-set NOT_FOUND=
-set PATH_DIRS=
-
-if "%PROCESSOR_ARCHITECTURE%" == "AMD64" (
-  call :which MSYSGIT       "%ProgramFiles(x86)%\Git\bin" git.exe
-) else (
-  call :which MSYSGIT       "%ProgramFiles%\Git\bin"  git.exe
-)
-call :which QTMINGW       "C:\QtSDK\mingw\bin"                  mingw32-make.exe
```
Comment (kubark42) [inline make/winx86/cmd/sh.cmd]: So glad to see all this go. This was always the confusing part of setting up a dev environment on Windows.

File: make/tools.mk (hunk 1 above)
Comment (mlyle) [inline make/tools.mk]: It's included with the mingw64 subset in the git install mentioned on the wiki.

File: make/tools.mk (hunk 2 above)
Comment (mlyle) [inline make/tools.mk]: OK.  But I can't make this atomic / have the intermediate versions work.  The old version of the openssl_install stuff didn't work, and the 1_0_2c.exe doesn't exist anymore on the upstream site.  So there's no intermediate version of this that will work...

File: make/winx86/cmd/sh.cmd (hunk 3 above)
Comment (mlyle) [inline make/winx86/cmd/sh.cmd]: Probably the biggest win is you end up in a real terminal this way, not the "command prompt" with broken copy and paste, etc.

File: make/tools.mk (hunk 2 above)
Comment (mlyle) [inline make/tools.mk]: Broken into another commit.

**Comment by mlyle** [comment]
@kubark42 -- glad you like it, and thanks for the review

File: make/tools.mk (hunk 1 above)
Comment (kubark42) [inline make/tools.mk]: Brilliant.

File: make/tools.mk (hunk 2 above)
Comment (kubark42) [inline make/tools.mk]: Thanks.

P.S. I get where you're coming from. There's a good philosophical question about whether every git commit should be able to compile and work, or whether git commits which break-- but then fix-- the build are okay.

I tend to want to accomplish the former, but I don't hesitate to do the latter. In my commit messages, you'll find lots of ```*** DOES NOT COMPILE***```. It seems to work for me, but to each his own...

File: make/winx86/cmd/sh.cmd (hunk 3 above)
Comment (kubark42) [inline make/winx86/cmd/sh.cmd]: <3

**Comment by PTDreamer** [comment]
I have promised myself not to participate anymore in anything code related here, but isn't the .bash_profile file missing?

**Comment by mlyle** [comment]
@PTDreamer thanks for spotting that.  When I reset to split up the commit I forgot to readd the bash_profile file.

And I'm sorry to hear about that promise.  We got crossways on those couple of PRs but your expertise and effort is valuable.

(I'm just about to look at that crash reporting PR again to see if I can make it fully work)

**Comment by kubark42** [comment]
Can a Windows user test this code? Or @mlyle can you test it on a Windows VM? It looks complete and ready to merge, but it'd be nice to get confirmation from a virgin platform.

**Comment by mlyle** [comment]
I built this up on a new Windows machine originally with no pre-existing install.  I'm planning on re-winding and getting my main windows dev platform set up this way, but it's not a clean slate.

**Comment by kubark42** [comment]
I'll be honest, there's a lot of room to make the current setup better, and little room to make it worse. The wiki instructions were sufficiently outdated to make me wonder if anyone used them at all.

It's the weekend, so maybe we're in luck and someone can give this a quick test. Otherwise, LGTM and I'll merge tomorrow.

**Comment by tracernz** [comment]
I tested this on a clean Windows 10 VM using the recommended version of git from the link in our wiki (2.5.3) using the default recommended terminal (MinTTY). It doesn't load the bash profile (copied to ~/.bash_profile as per wiki) and thus mingw32-make.exe never ends up on PATH.

Also, why OpenOCD 0.4.0 in the bash profile? 0.9.0 is the current version and I think our make openocd_install target builds 0.8.0.

**Comment by mlyle** [comment]
@tracernz  hmmm. Interesting.  It loads it for me.

0.4.0 in the bash profile probably isn't appropriate, you're right.  It was copied from the existing/wiki sh.cmd and that functionality wasn't tested.

**Comment by mlyle** [comment]
I am launching mintty through the "git bash" program shortcut ("start menu item")-- is that how you're launching it?

**Comment by mlyle** [comment]
image

**Comment by tracernz** [comment]
Strange, I was launching by right clicking in my Tau Labs directory and choosing Git bash. I tried the start menu shortcut with the same result:
```
Mike@DESKTOP-8ENQRJI MINGW64 ~
$ echo $PATH
/c/Users/Mike/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/mingw64/bin:/usr/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/cmd:/mingw64/bin:/usr/bin:/usr/bin/vendor_perl:/usr/bin/core_perl:/C/Python27

Mike@DESKTOP-8ENQRJI MINGW64 ~
$ ls ~/.bash_profile
/c/Users/Mike/.bash_profile

Mike@DESKTOP-8ENQRJI MINGW64 ~
$ echo $SHELL
/usr/bin/bash

Mike@DESKTOP-8ENQRJI MINGW64 ~
$ alias make
alias make='mingw32-make'

Mike@DESKTOP-8ENQRJI MINGW64 ~
$ which make
which: no make in (/c/Users/Mike/bin:/mingw64/bin:/usr/local/bin:/usr/bin:/bin:/mingw64/bin:/usr/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/cmd:/mingw64/bin:/usr/bin:/usr/bin/vendor_perl:/usr/bin/core_perl:/C/Python27)
```
I'll look into it further tomorrow.

**Comment by mlyle** [comment]
OK, this is dumb, and unique to mingw's /etc/profile.  You need to chmod 755 .bash_profile for it to work.  Creating the file locally via vi makes it executable; checking it out from git doesn't.  Updating wiki.

**Comment by kubark42** [comment]
@tracernz, can you confirm this works for you?

@mlyle, I hear your pain about the permissions. I've had this with other git repos as well. Were you able to reproduce @tracernz's problem and the fix comes via the updated wiki?

I'm looking forward to merging this upon confirmation.

**Comment by tracernz** [comment]
@kubark42 Sorry, been really busy lately (final year project presentation yesterday etc.). I'll take a look this afternoon. :)

Date: Fri, 2 Oct 2015 05:43:55 -0700

**Comment by tracernz** [comment]
Okay, the .bash_profile now loads and I've tested this. I used default paths for everything and Qt ended up at /C/Qt/Qt5.5.0 rather than /C/Qt so once I changed that everything appears to be working okay. Odd that the default path is different though? This may be a case of offline installer vs. online installer (I use offline to avoid re-downloading Qt everytime I set it up).

**Comment by kubark42** [comment]
~~~@tracernz That's a very easy problem to resolve...~~~

Nevermind, I see right now that @mlyle has already provided a fix. I'll merge this once Jenkins gives the go-ahead.

**Comment by tracernz** [comment]
@kubark42: Yep, I've just tested @mlyle's fix and it's good. :)

**Comment by kubark42** [comment]
Looking forward to a new era of Windows friendliness. Merged.
